util/csvresponse.py

```diff
@@ -48,18 +51,66 @@ def to_csv_files(self, path):
         file_paths = self._create_files(base_path)
         return json.dumps({"code": 200, "message": str(file_paths)}, indent=2)
 
+    def _create_filter_list(self,keys):
+        '''
+        generates a set of parameters to filter (remove) from the package
+        :param keys: list of keys from the dataset.
+        :return: set containing keys to filter from the dataset.
+        '''
+        # default parameters -- move to class variable?
+        default = ['time', 'deployment', 'lat', 'lon']
+
+        # initialize param list to include requested and default parameters
+        missing_params = []
+        params_to_include = []
+        # check for and remove any missing params from the requested list
+        for param in self.stream_request.requested_parameters:
+            if param.name not in keys:
+                missing_params.append(param.name)
+            else:
+                params_to_include.append(param.name)
+        if missing_params:
+            log.warning('one or more selected parameters (%s) not found in the dataset',missing_params)
+
+        # add the default params to the inclusion list
+        params_to_include.extend(default)
+
+        # Determine if there is interpolated pressure parameter. if so include it
+        pressure_params = [(sk,param) for sk in self.stream_request.external_includes
+                            for param in self.stream_request.external_includes[sk]
+                            if param.data_product_identifier == PRESSURE_DPI]
+        if pressure_params:
+            params_to_include.append(INT_PRESSURE_NAME)
+
+        # create the list of fields to remove from the dataset
+        # start with fields we never want to include
+        drop = {'id', 'annotations'}
+        for key in keys:
+            # remove any "extra" keys while keeping qc params and removing 'provenance' params
+            if (key not in params_to_include and not self._is_qc_param(key)) or 'provenance' in key:
+                drop.add(key)
+        return drop
+
+    def _is_qc_param(self,param):
```

Comment:
After plugging your code to PyCharm I see that this function should be static. Please use PyCharm or other PEP-8 compliant editor.

Reply:
My PyCharm wasn't telling me that, but I will make the change.